cnudie/iter.py

```diff
@@ -1,22 +1,41 @@
+import collections.abc
 import dataclasses
-import typing
+import enum
 
-import gci.componentmodel as cm
 import cnudie.retrieve
+import dso.labels
+import gci.componentmodel as cm
+
+
+class NodeReferenceType(enum.StrEnum):
+    COMPONENT_REFERENCE = 'componentReference'
+    EXTRA_COMPONENT_REFS_LABEL = dso.labels.ExtraComponentReferencesLabel.name
+
+
+@dataclasses.dataclass(frozen=True)
+class NodePathEntry:
+    component: cm.Component
+    reftype: NodeReferenceType = NodeReferenceType.COMPONENT_REFERENCE
 
 
 @dataclasses.dataclass
 class Node:
-    path: tuple[cm.Component]
+    path: tuple[NodePathEntry]
 
     @property
     def component(self):
-        return self.path[-1]
+        return self.path[-1].component
 
     @property
     def component_id(self):
         return self.component.identity()
 
+    def __post_init__(self):
+        self.path = tuple(
```

Comment:
shouldn't we rather expect callers to pass correct values (we might check + raise ValueError if values are incorrect)?